### ISSUE TITLE
Training of RTDETRv2 with multi-gpus is hanging 

### ISSUE THREAD

**Issue by VimukthiRandika1997** [lyuwenyu/RT-DETR#420]
We trained custom rtdetrv2 models using multi-gpu setting. With single gpu training it works fine. But when we utilized multi-gpus training is just hanging in the first epoch for a longer time. We haven't faced this kind of an issue with rtdetrv1.

Training details:
- Dataset size: 20, 000 images (both training and validation split)
- GPUs - 2 V100 gpus
- other parameters were set to default ones expect the batch size

Thank you for your awesome work!
@lyuwenyu

**Comment by 464029120** [comment]
I encountered the same problem

**Comment by afroserom** [comment]
I'm facing the same problem. Did you find a solution by any chance? @VimukthiRandika1997

**Comment by yenchi1222** [comment]
In my experiments, it caused by the nan result of loss_labels_vfl. It seems the loss is not well defined. 
A simple solution is to add
`
if torch.isnan(loss): 
    return {'loss_vfl' : torch.tensor(0., device = loss.device, requires_grad = False)}
`
after line 130 in rtdetrv2_pytorch/src/nn/criterion/det_criterion.py.

**Comment by bryan-pakulski** [comment]
I think I am experiencing this issue - 4x V100 in AWS, even with stock batch / worker counts training freezes on the first epoch, can see when it happens when monitoring gpu usage as it spikes to 100% on each device

I tried your suggestion @yenchi1222 but it didn't seem to work in my case - were you seeing these same symptoms?

**Comment by yenchi1222** [comment]
@bryan-pakulski I used to face the same issue. After I modified the loss, the issue had been solved. Maybe you could try not to use the ``amp'' argument.  If that doesn't help, try to run the code by python instead of torchrun command with one gpu for debugging.

**Comment by bryan-pakulski** [comment]
Cheers,

Will do some do some digging

**Comment by bryan-pakulski** [comment]
Just as an update on this it seems it could possibly be hardware related

I have set up a local training server using multiple rtx 3060's and that is able to train using multi gpu without any hangs

It's only on aws with 2+ V100 gpu's that I experience problems

Not spending anymore time looking into it, aws compute is expensive, and local compute has been much easier to use
